documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -11,203 +11,652 @@ Proposal API for backend-agnostic, refactored Input via keyboards, mice, and con
 - [ ] Implemented
 
 # Design Decisions
-- This proposal refactors the internals of the Input API to be less tied to a single backend, and instead allow multiple backends to be added and removed from an input context at will.
+- This proposal, while aiming to keep the same look and feel as the 2.0 API, is a complete redesign of the Input API with the following goals:
+    - Input backends can be added and removed at will to a central input context object.
+    - All input devices, regardless of what backends they come from, can be accessed from a central input context object.
+    - Should the Silk.NET team design more APIs for more devices (for example VR hands using OpenXR) in the future, this can be added in a non-breaking way.
+    - All input devices instead of exposing properties only expose a single `State` property, with additional properties and/or methods for writable state (i.e. vibration motors, mouse position). This also addresses comments from the Working Group on previous iterations of this proposal, where the Working Group expressed interest in the state being immutable and trivially capturable.
+    - Ensuring the input API is accessible and easy to use.
+    - Addressing APIs that the userbase frequently had issues/distaste with. For example, events are no longer on each individual device, rather they are aggregated in a list and accessed from a single entry-point.
```

Comment:
I like this for devices like `IMouse` and `IKeyboard`, which I don't think you could ever have two of, but what about 'IGamepad' or 'IJoystick'? You can connect two gamepads to a single PC for a locally-multiplayer game, does it make sense to keep the events for all unified on these too?

Reply:
Yeah, but with all of the events you have a "sender" as the first argument telling you which IDevice caused the event to happen